src/node/catchup.ml

```diff
@@ -95,7 +95,13 @@ let catchup_tlog (type s) ~tls_ctx ~tcp_keepalive ~stop other_configs ~cluster_i
       tlog_coll # log_value_explicit i value ~sync:false None >>= fun _ ->
       stop_fuse stop
     in
-    client # iterate current_i f tlog_coll ~head_saved_cb 
+    client # iterate current_i
+           f
+           tlog_coll
+           ~head_saved_cb
+    >>= fun fs_changed ->
+    if fs_changed then tlog_coll # invalidate();
```

Comment:
doing the invalidate could/should be considered a responsibility of `remote_nodestream # iterate`?
(also then there's no need to change the signature)